src/System.Private.CoreLib/shared/System/Collections/Generic/List.cs

```diff
@@ -339,11 +341,19 @@ public List<TOutput> ConvertAll<TOutput>(Converter<T, TOutput> converter)
             }
 
             List<TOutput> list = new List<TOutput>(_size);
-            for (int i = 0; i < _size; i++)
+            Debug.Assert((uint)_size <= (uint)list._items.Length);
+            list._size = _size;
+            // Use Span to avoid covarience checks from using List.
+            // We can use the ref overload as the list is a local and not subject to data races.
+            Span<TOutput> output = new Span<TOutput>(ref Unsafe.As<byte, TOutput>(ref list._items.GetRawSzArrayData()), _size);
```

Comment:
Is using the ref overload measurable?  If not, I'd prefer the simpler / visibly-safer code that doesn't require an explanatory comment.

Reply:
I was going for less asm in an instance method of a common generic class more than anything, but happy to switch over

Reply:
The asm code is not the whole footprint. The references to `Span<TOutput>` will add to the footprint as well, depending on the compilation model. This is unlikely to be a footprint improvement on average.

This should be looked at as throughput optimization. Do you see measurable difference in throughput?